tests/unit/test_workflow_files.py

```diff
@@ -1824,87 +1829,50 @@ def test_detect_both_flow_and_suite(tmp_path):
-        err: Type of exception if expected to get raised.
-        expected_file: Which file's path should get returned, when
-            symlink_suiterc_arg is FALSE (otherwise it will always be
-            flow.cylc, assuming no exception occurred).
-        symlink_suiterc_arg: Value of the symlink_suiterc arg passed to
-            check_flow_file().
-    """
+def test_detect_both_flow_and_suite_symlinked(tmp_path):
+    """Test flow.cylc symlinked to suite.rc together in dir is permitted."""
+    Path(tmp_path)
+    (tmp_path / WorkflowFiles.SUITE_RC).touch()
     flow_file = tmp_path.joinpath(WorkflowFiles.FLOW_FILE)
-    suiterc = tmp_path.joinpath(WorkflowFiles.SUITE_RC)
-    tmp_path.joinpath('other-path').touch()
-    if suiterc_exists:
-        suiterc.touch()
-    if flow_file_target:
-        flow_file.symlink_to(flow_file_target)
+    flow_file.symlink_to(WorkflowFiles.SUITE_RC)
+    try:
+        detect_both_flow_and_suite(tmp_path)
+    except WorkflowFilesError:
+        pytest.fail("Unexpected WorkflowFilesError")
+
+
+def test_flow_symlinked_elsewhere_and_suite_present(tmp_path):
+    """flow.cylc symlinked to suite.rc elsewhere, and suite.rc in dir raises"""
+    tmp_path = Path(tmp_path)
+    tmp_path.joinpath('some_other_dir').mkdir(exist_ok=True)
+    suite_file = tmp_path.joinpath('some_other_dir', WorkflowFiles.SUITE_RC)
+    suite_file.touch()
+    run_dir = tmp_path.joinpath('run_dir')
+    run_dir.mkdir(exist_ok=True)
+    flow_file = (run_dir / WorkflowFiles.FLOW_FILE)
+    flow_file.symlink_to(suite_file)
+    (run_dir / WorkflowFiles.SUITE_RC).touch()
+    forbidden = is_forbidden(flow_file)
+    assert forbidden is True
+    with pytest.raises(WorkflowFilesError) as exc:
+        detect_both_flow_and_suite(run_dir)
+        assert str(exc.value) == (
+            "Both flow.cylc and suite.rc files are present in the "
+            "source directory. Please remove one and try again. "
+            "For more information visit: "
+            "https://cylc.github.io/cylc-doc/latest/html/7-to-8/summary.html"
+            "#backward-compatibility"
+        )
 
-    caplog.set_level(logging.INFO, CYLC_LOG)
-    log_msg = ""
 
-    if err:
-        with pytest.raises(err):
-            check_flow_file(tmp_path, symlink_suiterc_arg)
-    else:
-        assert expected_file is not None  # otherwise test is wrong
-        result = check_flow_file(tmp_path, symlink_suiterc_arg)
-        if symlink_suiterc_arg is True:
-            assert flow_file.samefile(suiterc)
-            expected_file = WorkflowFiles.FLOW_FILE
-            if flow_file_target != WorkflowFiles.SUITE_RC:
-                log_msg = "Symlink created: flow.cylc -> suite.rc"
-        assert result == tmp_path.joinpath(expected_file)
-
-    if log_msg:
-        assert log_msg in caplog.messages
+def test_is_forbidden_symlink_returns_false_for_non_symlink(tmp_path):
+    """Test sending a non symlink path is not marked as forbidden"""
```

Comment:
Shouldn't a symlink also be not marked as forbidden?

Reply:
Yes, thanks @MetRonnie. Small adjustment and test added. I hope all bases are covered this time!